src/enclave/test/http.cpp

```diff
@@ -3,131 +3,223 @@
+  auto req = build_post_request(r);
   auto parsed = p.execute(req.data(), req.size());
 
-  sp.expect({r});
+  CHECK(!sp.received.empty());
+  const auto& m = sp.received.front();
+  CHECK(m.method == HTTP_POST);
+  CHECK(m.body == r);
 }
 
 TEST_CASE("Parsing error")
 {
-  std::string r("{}");
+  std::vector<uint8_t> r;
 
   StubProc sp;
   enclave::http::Parser p(HTTP_REQUEST, sp);
 
-  auto req = post(r);
+  auto req = build_post_request(r);
   req[6] = '\n';
   CHECK_THROWS_WITH(
     p.execute(req.data(), req.size()),
     "HTTP parsing failed: HPE_INVALID_HEADER_TOKEN: invalid character in "
     "header");
-  sp.expect({});
+
+  CHECK(sp.received.empty());
 }
 
 TEST_CASE("Partial request")
 {
-  std::string r("{}");
-
   StubProc sp;
   enclave::http::Parser p(HTTP_REQUEST, sp);
 
-  auto req = post(r);
+  const auto r0 = s_to_v(request_0);
+  auto req = build_post_request(r0);
   size_t offset = 10;
 
   auto parsed = p.execute(req.data(), req.size() - offset);
   CHECK(parsed == req.size() - offset);
   parsed = p.execute(req.data() + req.size() - offset, offset);
   CHECK(parsed == offset);
 
-  sp.expect({r});
+  CHECK(!sp.received.empty());
+  const auto& m = sp.received.front();
+  CHECK(m.method == HTTP_POST);
+  CHECK(m.body == r0);
 }
 
 TEST_CASE("Partial body")
 {
-  std::string r("{\"a_json_key\": \"a_json_value\"}");
-
   StubProc sp;
   enclave::http::Parser p(HTTP_REQUEST, sp);
 
-  auto req = post(r);
-  size_t offset = 10;
+  const auto r0 = s_to_v(request_0);
+  auto req = build_post_request(r0);
+  size_t offset = build_post_header(r0).size() + 4;
 
   auto parsed = p.execute(req.data(), req.size() - offset);
   CHECK(parsed == req.size() - offset);
-
   parsed = p.execute(req.data() + req.size() - offset, offset);
   CHECK(parsed == offset);
 
-  sp.expect({r});
+  CHECK(!sp.received.empty());
+  const auto& m = sp.received.front();
+  CHECK(m.method == HTTP_POST);
+  CHECK(m.body == r0);
 }
 
 TEST_CASE("Multiple requests")
 {
-  std::string r0("{\"a_json_key\": \"a_json_value\"}");
-  std::string r1("{\"another_json_key\": \"another_json_value\"}");
-
   StubProc sp;
   enclave::http::Parser p(HTTP_REQUEST, sp);
 
-  auto req = post(r0);
-  auto req1 = post(r1);
+  const auto r0 = s_to_v(request_0);
+  auto req = build_post_request(r0);
+  const auto r1 = s_to_v(request_1);
+  auto req1 = build_post_request(r1);
   std::copy(req1.begin(), req1.end(), std::back_inserter(req));
 
   auto parsed = p.execute(req.data(), req.size());
   CHECK(parsed == req.size());
 
-  sp.expect({r0, r1});
+  {
+    CHECK(!sp.received.empty());
+    const auto& m = sp.received.front();
+    CHECK(m.method == HTTP_POST);
+    CHECK(m.body == r0);
+  }
+
+  sp.received.pop();
+
+  {
+    CHECK(!sp.received.empty());
+    const auto& m = sp.received.front();
+    CHECK(m.method == HTTP_POST);
+    CHECK(m.body == r1);
+  }
+}
+
+TEST_CASE("Method parsing")
+{
+  StubProc sp;
+  enclave::http::Parser p(HTTP_REQUEST, sp);
+
+  bool choice;
```

Comment:
Minor: should this be set to `true` or `false` to start with?

Reply:
Sure, done!

Reply:
Thanks. It guarantees consistency between test runs, which I think is a good thing.